portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -30,9 +31,11 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
-    Key taskEntityKey = KeyFactory.createKey("Shows", id);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(taskEntityKey);
+    Query query = new Query("Shows");
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      datastore.delete(entity.getKey());
+    }
   }
 }
```

Comment:
Remember to keep newlines at end of file (can optionally add in this review or following reviews)

Reply:
Fix.